components/ipfs/ipfs_service.cc

```diff
@@ -432,6 +435,62 @@ void IpfsService::RemovePin(const std::vector<std::string>& cids,
       GetHeaders(gurl));
 }
 
+void IpfsService::RemovePinCli(std::set<std::string> cids,
+                               BoolCallback callback) {
+  if (cids.begin() == cids.end()) {
+    std::move(callback).Run(true);
+    return;
+  }
+
+  base::FilePath path = GetIpfsExecutablePath();
+  if (path.empty()) {
+    std::move(callback).Run(false);
+    return;
+  }
+
+  base::CommandLine cmdline(path);
+  cmdline.AppendArg("pin");
+  cmdline.AppendArg("rm");
+  cmdline.AppendArg("-r=true");
+
+  cmdline.AppendArg(*(cids.begin()));
```

Comment:
This looks scary. I guess it can only be called for valid CIDs and not an arbitrary string from the NFT's metadata (e.g `ipfs://;rm -rf /` ) ? Any reason you don't use `BraveWalletPinService::RemovePin` here instead?

Reply:
First of all BraveWalletPinService::RemovePin needs local node to be started.
Also it is more safe to delete all pins from the Kubo cache because there may be desync of pins listed in the IpfsLocalPinService and the Kubo's cache. Normally it is handled by GC task, but this GC task is not launched if the feature is disabled.

Reply:
I've added additional check whether provided cid is valid